Port `ae` commands to RzShell

Rot127 opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

Port the `ae` commands to RzShell.

**Remove**

- [x] `ae*` - It was marked as not working and had no direct advantage. With the transition to RzIL it becomes obsolete.


**To RzShell, but remove with ESIL**
- [x] `aepc`
- [x] `ae `
- [x] `ae??`
- [x] `aex`

**To RzShell, but port to RzIL**

- [x] `aeb`
- [x] `aef`
- [x] `aefa`
- [x] `aeA`
- [x] `aea`
- [x] `aeab`
- [x] `aeafj`
- [x] `aeafj`
- [x] `aeab`
- [x] `aeabj`

**To RzShell, but reimplement**
- [x] `aek`
- [x] `aek-`
- [x] `aeli`
- [x] `aeli`
- [x] `aelir`

**Last**

- [x] Remove `a*` (doesn't work)
- [x] Remove legacy `rz_cmd_analysis` handler + fixup in yaml files.

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

All green

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

closes https://github.com/rizinorg/rizin/issues/1533

Rot127 committed:
Remove ae* command.

It was marked as not working and had no direct advantage.
With the transition to RzIL it becomes obsolete.

Rot127 committed:
Port ae command to RzShell

Rot127 committed:
Port aepc to RzShell.

Rot127 committed:
Port aek commands to RzShell

Rot127 committed:
Fix help of ported ae commands

Rot127 committed:
Port aeb to RzShell

Rot127 committed:
Port aex command to RzShell

Rot127 committed:
Port aef commands to RzShell

Rot127 committed:
Port ae?? command to aeH

Rot127 committed:
Fix off by one error

Rot127 committed:
Port ael commands to RzShell.

Rot127 committed:
Port aea command to RzShell.

XVilka commented on librz/core/cmd_descs/cmd_analysis.yaml:
```diff
@@ -2402,3 +2402,139 @@ commands:
+        cname: analyze_esil_expr_help
+        args: []
+      - name: aeb
+        summary: Emulate current block.
```
"Emulate current block using ESIL VM" or something like that. Until ESIL completely removed we should make distinction clear.

XVilka commented on librz/core/cmd_descs/cmd_analysis.yaml:
```diff
@@ -2402,3 +2402,139 @@ commands:
+          - name: addr
+            type: RZ_CMD_ARG_TYPE_NUM
+      - name: aek
+        summary: SDB queries on ESIL.info.
```
Unnecessary dot?

XVilka commented on librz/core/cmd_descs/cmd_analysis.yaml:
```diff
@@ -2402,3 +2402,139 @@ commands:
+        summary: SDB queries on ESIL.info.
+        subcommands:
+          - name: aek
+            summary: Perform sdb query on ESIL.info.
```
Same

XVilka commented on librz/core/cmd_descs/cmd_analysis.yaml:
```diff
@@ -2402,3 +2402,139 @@ commands:
+                type: RZ_CMD_ARG_TYPE_STRING
+                default_value: "123*"
+          - name: aek-
+            summary: Resets the ESIL.info sdb instance.
```
Same.

XVilka commented on librz/core/cmd_descs/cmd_analysis.yaml:
```diff
@@ -2402,3 +2402,139 @@ commands:
+            cname: analyze_esil_sdb_reset
+            args: []
+      - name: aex
+        summary: Emulate instruction encoded in the given bytes.
```
Same for all those emulation via ESIL instructions

XVilka commented on librz/core/cmd/cmd_analysis.c:
```diff
@@ -6563,3 +6426,192 @@ RZ_IPI RzCmdStatus rz_analysis_data_trampoline_handler(RzCore *core, int argc, c
+		aea_mode |= 1;
+	}
+
+	// Don't ask me why this has to be done. I am just the janitor.
```
Lol

Rot127 replied:
It very much feels this way.

Rot127 committed:
Highlight ESIL is used for emulation in ae commands.

Rot127 committed:
Make ae parameter mandatory

Rot127 committed:
Remove a*

It executed not existing commands

Rot127 committed:
Remove legacy ae command handler.

Rot127 committed:
Fix ae tests by putting expressions in strings.

They contain special/illegal argument characters for the RzShell.

Rot127 committed:
Fix parameter order of aea command.

For the most common case people just pass a number.
Hence, it must be the first argument.

Rot127 committed:
Run yamllint
